src/components/Models/MultipleAnimationModel/MultipleAnimationModel.component.tsx

```diff
@@ -65,37 +61,28 @@ export const MultipleAnimationModel: FC<MultipleAnimationModelProps> = ({
     const mixer = mixerRef.current;
     const prevAction = activeActionRef.current;
     const newClip = loadedAnimations[activeAnimation];
+    const animationConfig = animations[activeAnimation];
 
-    if (!newClip || !mixer) return;
-    if (prevAction && prevAction.getClip().name === newClip.name && activeAnimation !== idleAnimation) return;
+    if (!newClip || !mixer || !animationConfig) return;
+    if (prevAction && prevAction.getClip().name === newClip.name) return;
 
     const newAction = mixer.clipAction(newClip);
-    if (activeAnimation === idleAnimation) {
-      newAction.setLoop(LoopRepeat, Infinity);
-    } else {
-      newAction.setLoop(LoopOnce, 1);
-      newAction.clampWhenFinished = true;
```

Comment:
Need to keep the  newAction.clampWhenFinished = true; so we crossfade from the last frame! All good otherwise

Reply:
ah yes, good catch

Reply:
added